src/archivematicaCommon/lib/archivesspace/client.py

```diff
@@ -0,0 +1,424 @@
+
+        :return list: A list of strings representing the record URIs for all children of the requested record.
+        """
+        def fetch_children(children):
+            results = []
+
+            for child in children:
+                results.append(child['record_uri'])
+
+                if child['has_children']:
+                    results.extend(fetch_children(child['children']))
+
+            return results
+
+        response = self._get(resource_id + '/tree')
+        tree = response.json()
+        return fetch_children(tree['children'])
+
+    def get_resource_component_children(self, resource_component_id):
+        """
+        Given a resource component, fetches detailed metadata for it and all of its children.
+
+        This is implemented using ArchivesSpaceClient.get_resource_component_children and uses its default options when fetching children.
+
+        :param string resource_component_id: The URL of the resource component from which to fetch metadata.
+        """
+        return self.get_resource_component_and_children(resource_component_id, 'resource')
+
+    def _format_dates(self, start, end=None):
+        if end is not None:
+            return "{}-{}".format(start, end)
+        else:
+            return start
+
+    def _fetch_dates_from_record(self, record):
+        if not record['dates']:
+            return ''
+        elif 'expression' in record['dates'][0]:
+            return record['dates'][0]['expression']
+        # use the first date, though there can be multiple sets
+        else:
+            start_date = record['dates'][0]['begin']
+            end_date = record['dates'][0].get('end')
+            return self._format_dates(start_date, end_date)
+
+    def _get_resources(self, resource_id, level=1, recurse_max_level=False, sort_by=None):
+        def format_record(record, level):
+            descend = recurse_max_level != level
+            level += 1
+
+            full_record = self._get(record['record_uri']).json()
+            dates = self._fetch_dates_from_record(full_record)
+
+            identifier = full_record['id_0'] if 'id_0' in full_record else full_record.get('component_id', '')
+
+            result = {
+                'id': record['record_uri'],
+                'sortPosition': level,
+                'identifier': identifier,
+                'title': record.get('title', ''),
+                'dates': dates,
+                'levelOfDescription': record['level']
+            }
+            if record['children'] and descend:
+                result['children'] = [format_record(child, level) for child in record['children']]
+                if sort_by is not None:
+                    kwargs = {'reverse': True} if sort_by == 'desc' else {}
+                    result['children'] = sorted(result['children'], key=lambda c: c['title'], **kwargs)
+            else:
+                result['children'] = False
```

Comment:
This is identical to the behaviour used by the previous Archivist's Toolkit code - I decided not to make any changes to the data format.